crates/router/src/core/pm_auth.rs

```diff
@@ -719,20 +741,41 @@ pub async fn retrieve_payment_method_from_auth_service(
         .map(common_utils::pii::Email::from)
         .get_required_value("email")?;
 
-    let payment_method_data = PaymentMethodData::BankDebit(BankDebitData::AchBankDebit {
-        billing_details: BankDebitBilling {
-            name: name.unwrap_or_default(),
-            email,
-            address: address_details,
-        },
-        account_number: masking::Secret::new(bank_account.account_number.clone()),
-        routing_number: masking::Secret::new(bank_account.routing_number.clone()),
-        card_holder_name: None,
-        bank_account_holder_name: None,
-        bank_name: None,
-        bank_type,
-        bank_holder_type: None,
-    });
+    let billing_details = BankDebitBilling {
+        name: name.unwrap_or_default(),
```

Comment:
Don't use `unwrap_or_default()`. Keep it a sensible default or keep it `None` if absent.